src/hooks/useToggledVersion.ts

```diff
@@ -21,3 +24,28 @@ export default function useToggledVersion(): Version {
       return Version.v3
   }
 }
+
+export function useToggleVersionCallback() {
+  const location = useLocation()
+  const search = useParsedQueryString()
+  const history = useHistory()
+  const version = useToggledVersion()
+
+  const linkDestination = useMemo(() => {
+    // pluck `use` out of object
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { use, ...rest } = search
+    const newUse = version === DEFAULT_VERSION ? { use: Version.v2 } : {}
+    return {
+      ...location,
+      search: `?${stringify({
+        ...rest,
+        ...newUse,
+      })}`,
+    }
+  }, [location, search, version])
+
+  return useCallback(() => {
+    history.replace(linkDestination)
```

Comment:
I like `replace` because if the user hits the back arrow in the browser to go back, we'll actually redirect them again to the better version. So pushing in this case has poor behavior imo.